incubator/hnc/internal/reconcilers/hierarchy_config.go

```diff
@@ -257,6 +300,10 @@ func (r *HierarchyConfigReconciler) markExisting(log logr.Logger, ns *forest.Nam
 }
 
 func (r *HierarchyConfigReconciler) syncParent(log logr.Logger, inst *api.HierarchyConfiguration, ns *forest.Namespace) {
+	if ns.IsExternal() {
+		return
+	}
```

Comment:
Shouldn't we call `ns.SetParent(nil)` in case there was a parent in the HC?

Reply:
Good catch, yes, I also added `ns.IsSub = false` in the early exit in `syncSubnamespaceParent()`.